etcdserver/server.go

```diff
@@ -1629,6 +1629,33 @@ func (s *EtcdServer) RemoveMember(ctx context.Context, id uint64) ([]*membership
 
 // PromoteMember promotes a learner node to a voting node.
 func (s *EtcdServer) PromoteMember(ctx context.Context, id uint64) ([]*membership.Member, error) {
+	resp, err := s.promoteMember(ctx, id)
+	if err != ErrNotLeader {
+		return resp, err
+	}
+
+	cctx, cancel := context.WithTimeout(ctx, s.Cfg.ReqTimeout())
+	defer cancel()
+	// forward to leader
+	for cctx.Err() == nil {
```

Comment:
if cctx.Err() is not `nil` return `cctx.Err()` ? I believe it can be either due to a deadline or a cancellation.

Reply:
I was following the same pattern as lease http. But I agree with you. I will update lease http error return as well.